info.go

```diff
@@ -259,6 +262,15 @@ func (pi *ProgramInfo) Runtime() (time.Duration, bool) {
 	return time.Duration(0), false
 }
 
+// RecursionMisses returns the total number of times the program was NOT called.
+//
+// It is INDEPENDENT on whether collection of statistics is enabled or not, but
+// it keeps the return values pattern as the other stats accessors and returns
+// bool (allways true) together with the counter value.
+func (pi *ProgramInfo) RecursionMisses() (uint64, bool) {
+	return pi.stats.recursionMisses, true
```

Comment:
Which kernel did this appear on though? The boolean is meant to indicate whether the kernel exposes this flag in general, not whether stats were enabled (IIRC).

Reply:
it got in v5.11 `9ed9e9ba2337 bpf: Count the number of times recursion was prevented`

I'm actually not sure how is the availability detected, both Runtime and RunCount
return values if `pi.stats != nil ` which seems to be the case always  AFAICS

I can keep the same checks as for the Runtime and RunCount

Reply:
> I'm actually not sure how is the availability detected, both Runtime and RunCount
return values if pi.stats != nil which seems to be the case always AFAICS

Can't remember who contributed it, but iirc they were planning to add a probe, and the bool was added to not have to break API later. Looks like the probe was never implemented, and looking back it hardly seems worth it to load a prog, call prog_run and verify if the run counter increases just so we can return that bool correctly.

In the case of RecursionMisses, that one feels unrealistic to try to reproduce reliably at all. I'd just drop the bool and add a comment about it being 5.11+.

Reply:
That breaks when using newProgramInfoFromProc. So unless we deprecate that we should just stick with the established pattern of returning a bool.

Reply:
ok, keeping the bool then, thanks

Reply:
pi.stats is nil and therefore the line panics.